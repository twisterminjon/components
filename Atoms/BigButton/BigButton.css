.bigbutton-button {
  width: 204px;
  height: 85px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  outline: none;

  color: #00295e;
  background-color: #dbe5f6;
  font-size: 18px;
  font-weight: 500;
  line-height: 22px;

  padding: 2px 34px;
  border-radius: 6px;

  flex-shrink: 0;
  flex-grow: 0;
}

.bigbutton-button:hover {
  color: #014092;
  background-color: #b6cdf3;
}
