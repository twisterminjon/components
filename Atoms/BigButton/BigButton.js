import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './BigButton.css'

export default class BigButton extends Component {
  static propTypes = {
    /** Label for the button */
    label: PropTypes.node.isRequired,
    /** Name used for aria-label */
    name: PropTypes.string.isRequired,
    /** Function for onClick */
    onClick: PropTypes.func,
  }
  static defaultProps = {
    onClick: () => {},
  }

  render() {
    const { label, name, onClick, ...rest } = this.props

    return (
      <button aria-label={name} className={`bigbutton-button`} onClick={onClick} {...rest}>
        {label}
      </button>
    )
  }
}
