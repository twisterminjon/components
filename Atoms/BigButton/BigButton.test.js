import React from 'react'
import BigButton from './BigButton'

describe('BigButton', () => {
  const mockFunction = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<BigButton onClick={mockFunction} label="button label" name="test" />)
    expect(wrapper).toMatchSnapshot()
  })
})
