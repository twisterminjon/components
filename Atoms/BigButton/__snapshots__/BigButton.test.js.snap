// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigButton matches the snapshot 1`] = `
<button
  aria-label="test"
  className="bigbutton-button"
  onClick={[MockFunction]}
>
  button label
</button>
`;
